### ISSUE TITLE
bug in smooth average for overlapping windows in removeLinesMovingWindow ?

### ISSUE THREAD

**Issue by awakenting** [VisLab/EEG-Clean-Tools#8]
Hello,
I was looking at the code for the line noise removal and if I understand correctly you are doing the smooth average of two consecutive windows in [this line](https://github.com/VisLab/EEG-Clean-Tools/blob/master/PrepPipeline/utilities/removeLinesMovingWindow.m#L79) where datafitwin should be the new window and datafitwin0 should be the old window. But in line 70 of the same function you already assign datafitwin to datafitwin0 so that the smooth average is computed with the first Noverlap samples and the last Noverlap samples both of the new window which would be wrong.
Here is the whole code snippet:
``` 
for n = 1:nw
        indx = winstart(n):(winstart(n) + Nwin - 1);
        datawin = data(indx);
         [datafitwin, f0Sig] = ...
            fitSignificantFrequencies(datawin, f0, lineNoise);
%         [datafitwin, f0Sig, FvalSig, aSig, fSig, sig] = ...
%             fitSignificantFrequencies(datawin, f0, lineNoise);
        datafitwin0 = datafitwin;
%         datafitWinSave(:, n) = datafitwin0;  % Debugging
%         f0SigSave{n} = f0Sig; % Debugging
%         FvalSigSave{n} = FvalSig;  % Debugging
%         aSigSave{n} = aSig;  % Debugging
%         fSigSave{n} = fSig;  % Debugging
%         sigSave{n} = sig;   % Debugging
        f0Mask = f0Mask | f0Sig;
        if n > 1
            datafitwin(1:Noverlap)= smooth.*datafitwin(1:Noverlap) + ...
                (1 - smooth).*datafitwin0((Nwin - Noverlap + 1):Nwin);
        end;
        datafit(indx, :) = datafitwin;
end
```
So I think the assignment to datafitwin0 should be after doing the smooth average.
I would appreciate if you could clarify this.

**Comment by VisLab** [comment]
I have checked out and you are correct. This bug propagated from the Chronux toolbox. I really appreciate your reporting and will update soon.

**Comment by VisLab** [comment]
Fixed in version 0.54
